src/collectors/diskspace.plugin/plugin_diskspace.c

```diff
@@ -617,7 +615,7 @@ static void diskspace_main_cleanup(void *ptr) {
     collector_info("cleaning up...");
 
     if (diskspace_slow_thread) {
-        nd_thread_join(*diskspace_slow_thread);
+        nd_thread_join(diskspace_slow_thread);
         freez(diskspace_slow_thread);
```

Comment:
`freez(diskspace_slow_thread);` is not needed

Reply:
You are right. I removed it.